domains/certificates/Query.API/API/MeasurementsSyncer/SlidingWindowService.cs

```diff
@@ -174,15 +193,37 @@ private static List<MeasurementInterval> FindMissingIntervals(MeteringPointTimeS
             {
                 if (IsCurrentMeasurementIndexInsideMissingInterval(currentMissingIntervalStart))
                 {
-                    AddMissingInterval(currentMissingIntervalStart!, UnixTimestamp.Create(currentMeasurement.DateFrom), missingIntervals);
+                    AddMissingIntervalUpToThreshold(currentMissingIntervalStart!, UnixTimestamp.Create(currentMeasurement.DateFrom), missingIntervals, minimumAgeThreshold);
                     currentMissingIntervalStart = null;
                 }
             }
         }
 
+        var remainingMissingIntervals = window.MissingMeasurements.Intervals
+            .Where(interval => interval.To > minimumAgeThreshold)
+            .Select(interval => interval.From < minimumAgeThreshold ? MeasurementInterval.Create(minimumAgeThreshold, interval.To) : interval)
+            .ToList();
+
+        missingIntervals.AddRange(remainingMissingIntervals);
+
         return missingIntervals;
     }
 
+    private static void AddMissingIntervalUpToThreshold(UnixTimestamp intervalStart, UnixTimestamp intervalEnd, List<MeasurementInterval> missingIntervals, UnixTimestamp threshold)
```

Comment:
This logic seems like something important to test :D Lots of if else and creating new stuff. Just FYI - I hate static keyword :D Would only use it if it is needed.

Reply:
Can this be resolved?